src/QsCompiler/CompilationManager/ProjectManager.cs

```diff
@@ -903,10 +904,10 @@ public ProjectManager(Action<Exception> exceptionLogger, Action<string, MessageT
             // TODO: allow to cancel this task via cancellation token?
             return this.load.QueueForExecutionAsync(() =>
             {
-                var loaded = projectLoader(projectFile, out ProjectInformation info);
+                var loaded = projectLoader(projectFile, out var info);
                 var existing = this.projects.TryRemove(projectFile, out Project current) ? current : null;
 
-                if (!loaded)
+                if (!loaded || info is null)
```

Comment:
With the check that you added to the `Loader` delagate, is the `info is null` condition needed here?

Reply:
Looks like the problem is that C# loses track of the `NotNullWhen` attribute when the return value is saved in a variable. I put the `projectLoader` call directly in the `if` and was able to avoid the extra `info is null` check.